pkg/controller/ecr/repository/zz_repository.go

```diff
@@ -0,0 +1,151 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package repository
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	svcapi "github.com/aws/aws-sdk-go/service/ecr"
+	svcsdkapi "github.com/aws/aws-sdk-go/service/ecr/ecriface"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	cpresource "github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/ecr/v1alpha1"
+	awsclient "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+const (
+	errUnexpectedObject = "managed resource is not an repository resource"
+
+	errCreateSession = "cannot create a new session"
+	errCreate        = "cannot create Repository in AWS"
+	errDescribe      = "failed to describe Repository"
+	errDelete        = "failed to delete Repository"
+)
+
+type connector struct {
+	kube client.Client
+}
+
+func (c *connector) Connect(ctx context.Context, mg cpresource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*svcapitypes.Repository)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+	cfg, err := awsclient.GetConfigV1(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
+	if err != nil {
+		return nil, err
+	}
+	sess, err := session.NewSession(cfg)
+	return &external{client: svcapi.New(sess), kube: c.kube}, errors.Wrap(err, errCreateSession)
```

Comment:
Based on https://aws.github.io/aws-sdk-go-v2/docs/migrating/ we can replace v1 `session.NewSession` with v2 `NewFromConfig`

```suggestion
	cfg, err := awsclient.GetConfig(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
	if err != nil {
		return nil, err
	}
	return &external{client: svcapi.NewFromConfig(cfg), kube: c.kube}, errors.Wrap(err, errCreateSession)
```